src/js/components/TabPanel/InfoWindow.js

```diff
@@ -22,67 +26,156 @@ export default class InfoWindow extends Component {
+      this.setState({
+        activeSelectedFeature
+      });
+    }
+  }
 
   previous = () => {
-    this.context.map.infoWindow.selectPrevious();
-    const selectedFeature = this.context.map.infoWindow.getSelectedFeature();
-    this.setState({
-      activeSelectedFeature: `{"name": "${selectedFeature.attributes[selectedFeature._layer.displayField] ? selectedFeature.attributes[selectedFeature._layer.displayField] : selectedFeature.attributes[selectedFeature._layer.objectIdField]}", "id": "${selectedFeature.attributes[selectedFeature._layer.objectIdField]}"}`
-    });
+    const {selectIndex} = this.state;
+    if (selectIndex > 0) {
+      const features = this.context.map.infoWindow.features;
+      const currentSelectedFeature = this.context.map.infoWindow.getSelectedFeature();
+      const matchedFeature = features.filter(feature => feature === currentSelectedFeature)[0];
+      const currentIndex = features.indexOf(matchedFeature);
+      let newIndex;
+      if (currentIndex - 1 > 0) {
+        newIndex = currentIndex - 1;
+      } else {
+        newIndex = currentIndex;
+      }
+      const newFeature = features[newIndex];
+      const newFeatureName = newFeature._layer.name;
+      this.context.map.infoWindow.select(newIndex);
+      const newSelectedFeature = layersCategories[newFeatureName];
+      this.setState({
+        activeSelectedFeature: `{"name": "${newSelectedFeature.name}", "count": "${newSelectedFeature.count}", "featuresList": "${newSelectedFeature.featuresList.map(feature => feature.attributes[feature._layer.objectIdField]).join()}"}`
+      });
+      mapActions.decreaseSelectIndex.defer();
+    }
   };
 
   next = () => {
-    this.context.map.infoWindow.selectNext();
-    const selectedFeature = this.context.map.infoWindow.getSelectedFeature();
-    this.setState({
-      activeSelectedFeature: `{"name": "${selectedFeature.attributes[selectedFeature._layer.displayField] ? selectedFeature.attributes[selectedFeature._layer.displayField] : selectedFeature.attributes[selectedFeature._layer.objectIdField]}", "id": "${selectedFeature.attributes[selectedFeature._layer.objectIdField]}"}`
-    });
+    const {selectIndex} = this.state;
+    const currentSelectedFeature = this.context.map.infoWindow.getSelectedFeature();
+    if (selectIndex < layersCategories[currentSelectedFeature._layer.name].count && selectIndex !== layersCategories[currentSelectedFeature._layer.name].count - 1) {
+      const features = this.context.map.infoWindow.features;
+      const matchedFeature = features.filter(feature => feature === currentSelectedFeature)[0];
+      const currentIndex = features.indexOf(matchedFeature);
+      let newIndex;
+      if (currentIndex + 1 > 0) {
+        newIndex = currentIndex + 1;
+      } else {
+        newIndex = currentIndex;
+      }
+      const newFeature = features[newIndex];
```

Comment:
@KaylaKremer Can we get rid of `all` of this and simply use the **matchedFeature** to access the `_layer` property and its `name` subproperty?!